### ISSUE TITLE
[ECR Public] [request]: please support the "data" field in content descriptors

### ISSUE THREAD

**Issue by tianon** [aws/containers-roadmap#2023]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Tell us about your request**
I want to push some OCI images, but they have the [optional `data` field](https://github.com/opencontainers/image-spec/pull/826) in their `config` descriptors, so I'm getting "405 Method Not Allowed" instead of a successfully pushed image.

**Which service(s) is this request for?**
ECR Public

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
I'm trying to push some verbatim OCI-spec-compliant images to ECR Public, but I can't push them as-is without modifying them (which would then change their digests and make them _different_ images).

**Are you currently working around this issue?**
Nope!  I've been apparently suffering from this problem for two months now, but only just today figured out why, and as a result, I likely won't fix it (I'll just stop pushing to ECR Public instead :see_no_evil:).

**Comment by jlbutler** [comment]
Hi @tianon! :wave:  We're looking into this now and should be able to give a timeline for addressing it soon. Thanks for the issue!

**Comment by rukumar333** [comment]
@jlbutler Checking in if there was any update on this?

**Comment by gauthamsam** [comment]
Hello, the work to support this ask in ECR Public is already underway. We plan to rollout the changes by 09/08.

**Comment by gauthamsam** [comment]
Hello, we have now launched support for "data" field in content descriptors in ECR Public. Please note that the launch also includes support for this field in layer and subject descriptors.

**Comment by tianon** [comment]
Nice, can confirm it appears to be working!  Thank you!

**Comment by adrianmace** [comment]
Just a heads up @jlbutler, the description in [the Public ECR page](https://gallery.ecr.aws/debian/debian) for this image still says the releases are blocked on this issue.

![image](https://github.com/aws/containers-roadmap/assets/5071859/db08d8e7-f65b-4f44-94d9-b28ff8ea53dd)

_Screenshot captured 27 Mar 2024 @ 09:41AM GMT+11_

**Comment by tianon** [comment]
... and now updated 👍
